RDKShell/RDKShell.cpp

```diff
@@ -544,12 +550,49 @@ namespace WPEFramework {
             }
             return exist;
         }
+                   
+       static void addClientRebootStatus(string client,PluginHost::IShell::state currentState,PluginHost::IShell::reason stateChangeReason)
+       {
+	 std::map <string,string>::iterator it;
+         bool isClientexist=false;
+         if(client == RESIDENTAPP_CALLSIGN )
+         {			 
+	  it=gClientsResetReason.find(client);
+
+	  if(it != gClientsResetReason.end())
+          {
+             isClientexist=true;
+          }
+
+          if((!isClientexist) && (currentState == PluginHost::IShell::ACTIVATED))
+          {
+              gRdkShellMutex.lock();
+              gClientsResetReason.insert(pair<string, string>(client,NORMALBOOT));
+              gRdkShellMutex.unlock();
+          }
+	  else if (((isClientexist)&& (currentState == PluginHost::IShell::DEACTIVATED) || (currentState == PluginHost::IShell::DESTROYED)))
+          {
+	    if(stateChangeReason == PluginHost::IShell::FAILURE)
+            {
+              it->second=CRASHBOOT;
+	    }
+            else
+	    {
+	      it->second=REQUESTBOOT;
+	    }
+	   
+         } 
+         else {}
+        }
+      }
 
         void RDKShell::MonitorClients::StateChange(PluginHost::IShell* service)
         {
             if (service)
             {
                 PluginHost::IShell::state currentState(service->State());
+				
+	        addClientRebootStatus(service->Callsign(),service->State(),service->Reason());
```

Comment:
add a check for resident app callsign here instead of within addClientRebootStatus